Tests/DiscoveryV2Tests/DiscoveryV2Tests.swift

```diff
@@ -0,0 +1,828 @@
+/**
+ * (C) Copyright IBM Corp. 2016, 2019.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+// swiftlint:disable function_body_length force_try force_unwrapping file_length
+
+import XCTest
+import Foundation
+// Do not import @testable to ensure only public interface is exposed
+import DiscoveryV2
+
+class DiscoveryTests: XCTestCase {
+    
+    private var discovery: Discovery!
+    private var projectID: String!
+    private var collectionID: String!
+    private let timeout: TimeInterval = 30.0
+    
+    // MARK: - Test Configuration
+    
+    override func setUp() {
+        super.setUp()
+        continueAfterFailure = false
+        instantiateDiscovery()
+    }
+    
+    func instantiateDiscovery() {
+        // TODO: CP4D authenticator
+        let username = WatsonCredentials.DiscoveryV2CPDUsername
+        let password = WatsonCredentials.DiscoveryV2CPDPassword
+        let url = WatsonCredentials.DiscoveryV2CPDURL
+        
+        let authenticator = WatsonCloudPakForDataAuthenticator.init(username: username, password: password, url: url)
+        authenticator.disableSSLVerification()
+        
+        discovery = Discovery(version: "2019-11-30", authenticator: authenticator)
+        
+        discovery.serviceURL = WatsonCredentials.DiscoveryV2ServiceURL
+        
+        discovery.defaultHeaders["X-Watson-Learning-Opt-Out"] = "true"
+        discovery.defaultHeaders["X-Watson-Test"] = "true"
+        
+        discovery.disableSSLVerification()
+        
+        projectID = WatsonCredentials.DiscoveryV2TestProjectID
+        collectionID = WatsonCredentials.DiscoveryV2TestCollectionID
+    }
+    
+    func loadDocument(name: String, ext: String) -> Data? {
+        #if os(Linux)
+        let url = URL(fileURLWithPath: "Tests/DiscoveryV1Tests/Resources/" + name + "." + ext)
+        #else
+        let bundle = Bundle(for: type(of: self))
+        guard let url = bundle.url(forResource: name, withExtension: ext) else { return nil }
+        #endif
+        let data = try? Data(contentsOf: url)
+        return data
+    }
+    
+    // MARK: - Test Definition for Linux
+    
+    static var allTests: [(String, (DiscoveryTests) -> () throws -> Void)] {
+        let tests: [(String, (DiscoveryTests) -> () throws -> Void)] = []
```

Comment:
Shouldn't you have the tests listed here to run on Linux?

Reply:
We don't have a CPD environment capable of running tests on Linux (due to the fact that we do not support self signed certs in the Swift SDK for CPD)

Reply:
Is Discovery CPD only?  I thought it was going to be available as a public service as well.